### PR TITLE
Add support for Gradle

### LINKED ISSUE #3353: Gradle Syntax Support
**Motivation**
We often write Gradle scripts in docs, which is similar to groovy but different.

**Description**
To highlight Gradle build scripts.

**Alternatives**
N/A

### PR DESCRIPTION
**Pull request by zeabdelkhalek**
This resolves #3353

### CODE REVIEW

File: examples/prism-gradle.html [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+string"""
+'''Multi-line
+string'''
+"String /containing/ slashes"
+</code></pre>
+
+<h2>Slashy strings (regex)</h2>
+<pre><code>/.*foo.*/
+/regex"containing quotes"/
+$/.*"(.*)".*/(.*)/$</code></pre>
```
Comment (RunDevelopment) [inline examples/prism-gradle.html:19]: Is that even correct?

I know that our tests force you to add an example page, but you can also just add a small code snippet as an example. We do that for a lot of languages. The full example at the bottom of the page is enough.

**Review by RunDevelopment — CHANGES_REQUESTED** [review]
Thank you for the PR @zeabdelkhalek!

It seems like you based this language of Groovy. That's obviously a good choice, but please adjust the tests. It seems like most of them have just been copied over, so please adjust them.

Also, code style. Please copy the code style from languages.

**Comment by zeabdelkhalek** [comment]
@RunDevelopment I have made the necessary changes, could you check again please?

File: components.json [hunk 2]
```diff
@@ -1414,10 +1387,7 @@
 		},
 		"tremor": {
 			"title": "Tremor",
-			"alias": [
-				"trickle",
-				"troy"
-			],
+			"alias": ["trickle", "troy"],
```
Comment (RunDevelopment) [inline components.json:1390]: Did ESLint do this? Sorry. Please don't re-format this file.

File: components/prism-gradle.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+			},
+			expression: {
+				pattern: /[\s\S]+/,
+				inside: null,
+			},
+		},
+	};
+
+	Prism.languages.gradle = Prism.languages.extend('clike', {
+		string: {
```
Comment (RunDevelopment) [inline components/prism-gradle.js:18]: Please surround token names with single quotes, even if they don't contain hyphens.

File: tests/languages/gradle/keyword_feature.test [hunk 4]
```diff
@@ -0,0 +1,27 @@
+implementation
+val tasks dependencies project plugin plugins
+repository repositories apply def sourceSets import
```
Comment (RunDevelopment) [inline tests/languages/gradle/keyword_feature.test:3]: CI fails because the `if` and `else` keywords aren't tested. Please add them.

**Review by RunDevelopment — CHANGES_REQUESTED** [review]

**Comment by zeabdelkhalek** [comment]
@RunDevelopment fixed, can you check again?

File: components.json [hunk 5]
```diff
@@ -1729,4 +1734,4 @@
 			"owner": "Golmote"
 		}
 	}
-}
+}
```
Comment (RunDevelopment) [inline components.json:1737]: Please add a new line at end here.

File: components/prism-gradle.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+			pattern: /#!.+/,
+			alias: 'comment',
+			greedy: true,
+		},
+		'interpolation-string': {
+			pattern:
+				/"""(?:[^\\]|\\[\s\S])*?"""|(["/])(?:\\.|(?!\1)[^\\\r\n])*\1|\$\/(?:[^/$]|\$(?:[/$]|(?![/$]))|\/(?!\$))*\/\$/,
+			greedy: true,
+			inside: {
+				interpolation: interpolation,
```
Comment (RunDevelopment) [inline components/prism-gradle.js:44]: Please add quotes here as well.

File: components/prism-gradle.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
+(function (Prism) {
+	var interpolation = {
+		pattern: /((?:^|[^\\$])(?:\\{2})*)\$(?:\w+|\{[^{}]*\})/,
+		lookbehind: true,
+		inside: {
+			'interpolation-punctuation': {
+				pattern: /^\$\{?|\}$/,
+				alias: 'punctuation',
+			},
+			expression: {
```
Comment (RunDevelopment) [inline components/prism-gradle.js:10]: And here too please.

File: examples/prism-gradle.html [hunk 8]
```diff
@@ -0,0 +1,56 @@
+<h2>Full example</h2>
+<pre><code>
+    apply plugin: "java"
```
Comment (RunDevelopment) [inline examples/prism-gradle.html:3]: Could you please un-indent the code once? It's weird that all lines start with 4 spaces.

Also, please remove the line break after `<code>` and make it `...<code>apply ...`. Right now, the code block starts with an empty line, and that's a little weird.

**Review by RunDevelopment — CHANGES_REQUESTED** [review]
I found a few minor things, but otherwise it's good to go.

Also, please run `npm run build` to update the minified files.

File: components.json (hunk 5 above)
Comment (zeabdelkhalek) [reply components.json]: Fixed

File: components/prism-gradle.js (hunk 6 above)
Comment (zeabdelkhalek) [reply components/prism-gradle.js]: Fixed

File: components/prism-gradle.js (hunk 7 above)
Comment (zeabdelkhalek) [reply components/prism-gradle.js]: Fixed

File: examples/prism-gradle.html (hunk 8 above)
Comment (zeabdelkhalek) [reply examples/prism-gradle.html]: Fixed

**Comment by zeabdelkhalek** [comment]
@RunDevelopment can you check again?

**Comment by RunDevelopment** [comment]
Thank you for contributing @zeabdelkhalek!

**Pull request merged**